ui/imports/shared/popups/RequestToJoinCommunity.qml

```diff
@@ -0,0 +1,74 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.1
+import QtQml.Models 2.14
+
+import utils 1.0
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Popups.Dialog 0.1
+
+StatusDialog {
+    id: root
+
+    property string name
+    property string introMessage
+    property url imageSrc
+    signal selectAddressesToShare()
+    signal shareSelectedAddresses()
+
+    padding: 0
+    title: qsTr("Request to join %1").arg(name)
+
+    footer: StatusDialogFooter {
+        rightButtons: ObjectModel {
+            StatusFlatButton {
+                borderColor: Theme.palette.baseColor2
+                text: qsTr("Select addresses to share")
+                onClicked: {
+                    root.selectAddressesToShare();
+                }
+            }
+            StatusButton {
+                text: qsTr("Share your addresses to join")
+                icon.name: "addresses"
```

Comment:
This icon should be one of:

![image](https://github.com/status-im/status-desktop/assets/5377645/49c0e22e-3e77-44cd-9705-e6b94e650c69)